src/install-script/installer.ts

```diff
@@ -0,0 +1,200 @@
+import * as archiver from "archiver";
+import { ChildProcess, exec, spawn } from "child_process";
+import { createWriteStream, ensureDir, remove } from "fs-extra";
+import { join as joinPath } from "path";
+
+export enum ScriptMessageType {
+  MINOR,
+  MAJOR,
+  ERROR,
+  COMPLETE
+}
+
+export enum InfoSource {
+  NPM = "npm",
+  VERDACCIO = "verdaccio",
+  OFFLINE_NPM = "offline-npm",
+  ARCHIVER = "archiver"
+}
+
+export interface IScriptMessage {
+  type: ScriptMessageType;
+  message: string;
+  source: InfoSource;
+}
+
+// Utility to make using callbacks with IScriptMessages easier
+const makeOutputter = (
+  callback: (data: IScriptMessage) => void,
+  type: ScriptMessageType
+) => (message: string, source: InfoSource = InfoSource.OFFLINE_NPM) => {
+  callback({
+    message,
+    source,
+    type
+  });
+};
+
+/**
+ * Installs an npm cache through Verdaccio and bundles the results in a zip file.
+ * @param packages the list of package names to install
+ * @param onInfo callback function for info messages
+ * @param onError callback function for error messages
+ * @param onComplete callback function for when the installation is complete
+ * @param outputPath the directory in which storage.zip should be placed
+ * @param verdaccioPath the directory in which the verdaccio source is located (usually node_modules)
+ */
+export const installCache = (
+  packages: string[],
+  onInfo: (data: IScriptMessage) => void,
+  onError: (data: IScriptMessage) => void,
+  onComplete: (data: IScriptMessage) => void,
+  outputPath: string = joinPath(__dirname, "out"),
+  verdaccioPath: string = joinPath(
+    __dirname,
+    "..",
+    "..",
+    "node_modules",
+    "verdaccio"
+  )
+): void => {
```

Comment:
Suggestion: it's convention these days to use `Promise`s, ideally with `async`/`await`, to manage asynchronous behavior. I think this would be more reusable if it returned a `Promise<Results>` _(where `Results` describes what happened)_. It's a bit hard to use `async`/`await` with traditional Node APIs that use callbacks / `.on`, but you can probably get rid of `onError` and `onComplete` by using a `resolve`.